test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java

```diff
@@ -166,66 +164,87 @@
+          .setGetURLSegmentsResult(true).build();
   private static final ContentExchange SITES_SITECOLLECTION_S_CONTENT_EXCHANGE
-      = new ContentExchange(ObjectType.SITE, null, null, null, true, false,
-          null, loadTestString("sites-SiteCollection-s.xml"));
+      = new ContentExchange.Builder().setObjectType(ObjectType.SITE)
+          .setRetrieveChildItems(true).setSecurityOnly(false)
+          .setGetContentResult(loadTestString("sites-SiteCollection-s.xml"))
+          .build();
   private static final ContentExchange SITES_SITECOLLECTION_SC_CONTENT_EXCHANGE
-      = new ContentExchange(ObjectType.SITE_COLLECTION, null, null, null,
-          true, false, null, loadTestString("sites-SiteCollection-sc.xml"));
+      = new ContentExchange.Builder().setObjectType(ObjectType.SITE_COLLECTION)
+          .setRetrieveChildItems(true).setSecurityOnly(false)
+          .setGetContentResult(loadTestString("sites-SiteCollection-sc.xml"))
+          .build();
   private static final URLSegmentsExchange
       SITES_SITECOLLECTION_LISTS_CUSTOMLIST_URLSEG_EXCHANGE
-      = new URLSegmentsExchange(
+      = new URLSegmentsExchange.Builder().setStrURL(
           "http://localhost:1/sites/SiteCollection/Lists/Custom List"
-            + "/AllItems.aspx",
-          true, null, null, "{6F33949A-B3FF-4B0C-BA99-93CB518AC2C0}", null);
+            + "/AllItems.aspx").setGetURLSegmentsResult(true)
+          .setStrListID("{6F33949A-B3FF-4B0C-BA99-93CB518AC2C0}").build();
   private static final ContentExchange
       SITES_SITECOLLECTION_LISTS_CUSTOMLIST_L_CONTENT_EXCHANGE
-      = new ContentExchange(ObjectType.LIST,
-          "{6F33949A-B3FF-4B0C-BA99-93CB518AC2C0}", null, null, false, false,
-          null, loadTestString("sites-SiteCollection-Lists-CustomList-l.xml"));
+      = new ContentExchange.Builder().setObjectType(ObjectType.LIST)
+          .setObjectId("{6F33949A-B3FF-4B0C-BA99-93CB518AC2C0}")
+          .setRetrieveChildItems(false).setSecurityOnly(false)
+          .setGetContentResult(loadTestString(
+              "sites-SiteCollection-Lists-CustomList-l.xml"))
+          .build();
   private static final ContentExchange
       SITES_SITECOLLECTION_LISTS_CUSTOMLIST_F_CONTENT_EXCHANGE
-      = new ContentExchange(ObjectType.FOLDER,
-          "{6F33949A-B3FF-4B0C-BA99-93CB518AC2C0}", "", null, true, false,
-          null, loadTestString("sites-SiteCollection-Lists-CustomList-f.xml"));
+      = new ContentExchange.Builder().setObjectType(ObjectType.FOLDER)
+          .setObjectId("{6F33949A-B3FF-4B0C-BA99-93CB518AC2C0}")
+          .setFolderUrl("").setRetrieveChildItems(true).setSecurityOnly(false)
+          .setGetContentResult(loadTestString(
+              "sites-SiteCollection-Lists-CustomList-f.xml")).build();
   private static final URLSegmentsExchange
       SITES_SITECOLLECTION_LISTS_CUSTOMLIST_1_URLSEG_EXCHANGE
-      = new URLSegmentsExchange(
+      = new URLSegmentsExchange.Builder().setStrURL(
```

Comment:
This wrapping is not nice. I would move both setStrURL and setGetURLSegmentsResult to the next lines.

Reply:
Done